x509/src/certificate.rs

```diff
@@ -202,6 +202,36 @@ pub struct Certificate<'a> {
     pub signature: BitString<'a>,
 }
 
+// impl<'a> ::der::Decodable<'a> for Certificate<'a> {
+//     fn decode(decoder: &mut ::der::Decoder<'a>) -> ::der::Result<Self> {
+//         decoder.sequence(|decoder| {
+impl<'a> DecodeValue<'a> for Certificate<'a> {
+    fn decode_value(decoder: &mut Decoder<'a>, _length: Length) -> der::Result<Self> {
+        let tbs_certificate = decoder.decode()?;
+        let signature_algorithm = decoder.decode()?;
+        let signature = decoder.decode()?;
+        Ok(Self {
+            tbs_certificate,
+            signature_algorithm,
+            signature,
+        })
+        // })
+    }
+}
+
+impl<'a> ::der::Sequence<'a> for Certificate<'a> {
+    fn fields<F, T>(&self, f: F) -> ::der::Result<T>
+    where
+        F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
+    {
+        f(&[
+            &self.tbs_certificate,
+            &self.signature_algorithm,
+            &self.signature,
+        ])
+    }
+}
+
```

Comment:
That is possibly one that could revert to the macro, and we should try. Generally, places where I did not use the macros it's because the macros did not achieve the desired end. For some, I updated the macros, for some I did not. The macros could be revisited in most of those places (but this may be one that ought revert).

Reply:
I would propose that anywhere the macros don't work, an issue should be filed for the missing or broken functionality and a reference to the issue should be place in a comment next to the manual implementation.

Reply:
I can open issues for each manual deviation. That's not a problem.